### PR TITLE
Tests/fix for test on ibmc

### PR DESCRIPTION
**Pull request by anandrkskd**
Signed-off-by: anandrkskd <anandrkskd@gmail.com>

**What type of PR is this?**

/kind failing-test

**What does this PR do / why we need it**:
This PR fixes tests that fail for windows and mac

**Which issue(s) this PR fixes**:

Fixes - part of #4412
Fixes - part of #4408

**PR acceptance criteria**:

- [ ] Unit test 

- [ ] Integration test 

- [ ] Documentation 

- [ ] Update changelog

- [x] I have read the [test guidelines](https://github.com/openshift/odo/blob/master/docs/dev/test-architecture.adoc)

**How to test changes / Special notes to the reviewer**:

### CODE REVIEW

File: tests/integration/cmd_push_test.go [hunk 1]
```diff
@@ -118,7 +118,8 @@ var _ = Describe("odo push command tests", func() {
 			// create a file and folder then push
 			err := os.MkdirAll(filepath.Join(commonVar.Context, "tests"), 0750)
 			Expect(err).To(BeNil())
-			_, err = os.Create(filepath.Join(commonVar.Context, "README.md"))
+			testFile, err := os.Create(filepath.Join(commonVar.Context, "README.md"))
+			testFile.Close()
```
Comment (valaparthvi) [inline tests/integration/cmd_push_test.go:122]: Why do you close the file as soon as it is created?

File: tests/integration/component.go [hunk 2]
```diff
@@ -386,8 +386,13 @@ func componentTests(args ...string) {
...
 				// Was failing due to https://github.com/openshift/odo/issues/1969
-				helper.Cmd("odo", append(args, "create", "--s2i", "java:8", cmpName, "--project",
-					commonVar.Project, "--binary", filepath.Join(commonVar.Context, "sb.jar"))...).ShouldPass()
+				if runtime.GOOS == "darwin" {
+					helper.Cmd("odo", append(args, "create", "--s2i", "java:8", cmpName, "--project",
+						commonVar.Project, "--binary", filepath.Join("/private", commonVar.Context, "sb.jar"))...).ShouldPass()
+				} else {
+					helper.Cmd("odo", append(args, "create", "--s2i", "java:8", cmpName, "--project",
+						commonVar.Project, "--binary", filepath.Join(commonVar.Context, "sb.jar"))...).ShouldPass()
+				}
```
Comment (valaparthvi) [inline tests/integration/component.go:395]: Perhaps just put the filepath part in if-else logic? Something like this? -
```go
binaryFilePath = filepath.Join(commonVar.Context, "sb.jar"))
if runtime.GOOS == "darwin" {
    binaryFilePath = filepath.Join("/private", binaryFilePath)
}
helper.Cmd("odo", append(args, "create", "--s2i", "java:8", cmpName, "--project",						commonVar.Project, "--binary", binaryFilePath...).ShouldPass()
```

File: tests/integration/cmd_push_test.go (hunk 1 above)
Comment (mohammedzee1000) [reply tests/integration/cmd_push_test.go:122]: yea, do you mean `defer testFile.Close()`

Comment (anandrkskd) [reply tests/integration/cmd_push_test.go:122]: > Why do you close the file as soon as it is created?

we are not writing anything in the file, so just closed it.

> yea, do you mean
yeah, forgot to added `defer`

File: tests/integration/component.go (hunk 2 above)
Comment (anandrkskd) [reply tests/integration/component.go:395]: Thanks, This makes more sense

File: tests/integration/component.go [hunk 3]
```diff
@@ -385,9 +385,12 @@ func componentTests(args ...string) {
...
 
 			It("should not fail when --context is not set", func() {
-				// Was failing due to https://github.com/openshift/odo/issues/1969
+				binaryFilePath := filepath.Join(commonVar.Context, "sb.jar")
+				if runtime.GOOS == "darwin" {
+					binaryFilePath = filepath.Join("/private", binaryFilePath)
+				}
 				helper.Cmd("odo", append(args, "create", "--s2i", "java:8", cmpName, "--project",
-					commonVar.Project, "--binary", filepath.Join(commonVar.Context, "sb.jar"))...).ShouldPass()
+					commonVar.Project, "--binary", filepath.Join(binaryFilePath, "sb.jar"))...).ShouldPass()
```
Comment (valaparthvi) [inline tests/integration/component.go:393]:
```suggestion
					commonVar.Project, "--binary", binaryFilePath...).ShouldPass()
```

**Review by valaparthvi — CHANGES_REQUESTED** [review]

**Review by valaparthvi — APPROVED** [review]
Code lgtm.

File: tests/integration/component.go [hunk 4]
```diff
@@ -385,9 +385,12 @@ func componentTests(args ...string) {
...
 
 			It("should not fail when --context is not set", func() {
-				// Was failing due to https://github.com/openshift/odo/issues/1969
+				binaryFilePath := filepath.Join(commonVar.Context, "sb.jar")
+				if runtime.GOOS == "darwin" {
+					binaryFilePath = filepath.Join("/private", binaryFilePath)
+				}
 				helper.Cmd("odo", append(args, "create", "--s2i", "java:8", cmpName, "--project",
-					commonVar.Project, "--binary", filepath.Join(commonVar.Context, "sb.jar"))...).ShouldPass()
+					commonVar.Project, "--binary", binaryFilePath...).ShouldPass()
```
Comment (valaparthvi) [inline tests/integration/component.go:393]: Missed a closing bracket there.
```suggestion
			helper.Cmd("odo", append(args, "create", "--s2i", "java:8", cmpName, "--project",
					commonVar.Project, "--binary", binaryFilePath)...).ShouldPass()
```

File: tests/integration/component.go [hunk 5]
```diff
@@ -404,14 +407,12 @@ func componentTests(args ...string) {
 			It("should be valid if path is relative and includes ../", func() {
 				relativeContext := fmt.Sprintf("..%c%s", filepath.Separator, filepath.Base(commonVar.Context))
 				fmt.Printf("relativeContext = %#v\n", relativeContext)
```
Comment (valaparthvi) [inline tests/integration/component.go:409]: Do we need the print statement here?

Comment (anandrkskd) [reply tests/integration/component.go:409]: I guess we don't need it, I guess it was added long back by Tomas.

Comment (anandrkskd) [reply tests/integration/component.go:409]: I will remove and push it.

File: tests/integration/cmd_push_test.go (hunk 1 above)
Comment (prietyc123) [reply tests/integration/cmd_push_test.go:122]: @anandrkskd Could you please add `defer` here

**Review by prietyc123 — CHANGES_REQUESTED** [review]

File: tests/integration/cmd_push_test.go (hunk 1 above)
Comment (feloy) [reply tests/integration/cmd_push_test.go:122]: No, the `defer` is not necessary as we don't want to use the file after this point. But you should check the err before using `Close` on `testFile`

**Comment by anandrkskd** [comment]
@prietyc123 I think we need to close it there only because defer closes it in the end of the function and there are some operation that will require the file to be closed.

**Pull request merged**